File: server/skill_area/services/session.py
from jsonrpc.exceptions import JSONRPCDispatchException

from skill_area.base import Service, utils
from skill_area.models import User, Session

UserNotExistsException = JSONRPCDispatchException(code=0, message='The email or password is incorrect.')
SessionNotExistsException = JSONRPCDispatchException(code=1, message='The not exists.')


class SessionService(Service):
    def create(self, _, username, password):
        try:
            user = User.objects.get(username=username, password=utils.encode_password(password))
        except User.DoesNotExist:
            raise UserNotExistsException

        try:
            session = Session.objects.get(user=user)
        except Session.DoesNotExist:
            session = Session.objects.create(user=user)

        return {
            'access_key': str(session.access_key),
            'category': user.category
        }

    def get(self, session):
        if not session:
            raise SessionNotExistsException

        return {
            'access_key': str(session.access_key),
            'category': session.user.category
        }

File: client/src/services/session.js
import Service from '../base/Service.js'
import UserService from './user.js'

class SessionService extends Service {
  prefix = 'sessions';

  create = ({username, password}) => this._call('create', {username, password});
  get = () => this._call('get', {})
}

class SessionStorage {
  create({username, password}) {
    const user = UserService.data.filter(user => user.username === username && user.password === password);
    if (user.length === 0) {
      throw Error('login or password is uncorrect');
    }

    UserService.current = {
      email: user[0].email,
      username: user[0].username
    };
    return {access_id: 0}
  }

  get() {
    return {access_id: 0}
  }
}

export default new SessionService();